### ISSUE TITLE
[BUG] CommunityToolki.Maui.Camera MauiAppBuilder bootstrapper method cannot be applied in custom builder extension class

### ISSUE THREAD

**Issue by cam13469** [CommunityToolkit/Maui#2030]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

Normally in a MAUI app when one wants to use the CommunityToolkit they must call the bootstrapper method on the MauiAppBuilder object created in the `MauiProgram.cs` file:

```cs
// MauiProgram.cs
using Microsoft.Extensions.Logging;

using CommunityToolkit.Maui;

namespace MyNameSpace
{
    public static class MauiProgram
    {
        public static MauiApp CreateMauiApp()
        {
            var builder = MauiApp.CreateBuilder();
            builder
                .UseMauiApp<App>()
                .UseMauiCommunityToolkit()        // Bootstrap CommunityToolkit
                .ConfigureFonts(fonts =>
                {
                    fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                    fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
                });

#if DEBUG
    		builder.Logging.AddDebug();
#endif

            return builder.Build();
        }
    }
}
```

With the addition of the `CameraView` component in the `CommunityTookit.Maui.Camera` package, it is required to also register this package with the MauiAppBuilder object via a bootstrapping method:

```cs
// MauiProgram.cs
using Microsoft.Extensions.Logging;

using CommunityToolkit.Maui;

namespace MyNameSpace
{
    public static class MauiProgram
    {
        public static MauiApp CreateMauiApp()
        {
            var builder = MauiApp.CreateBuilder();
            builder
                .UseMauiApp<App>()
                .UseMauiCommunityToolkit()              // Bootstrap CommunityToolkit
                .UseMauiCommunityToolkitCamera()        // Bootstrap CommunityToolkit Camera
                .ConfigureFonts(fonts =>
                {
                    fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                    fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
                });

#if DEBUG
    		builder.Logging.AddDebug();
#endif

            return builder.Build();
        }
    }
}
```

This is all well and good, however my requirements enforce the need to apply these bootstrapping calls to the MauiAppBuilder object via an extension class. An extension class meeting these requirements may look something like this:

```cs
// MauiAppBuilderExtensions.cs
using CommunityToolkit.Maui;

namespace MyNameSpace;

public static partial class MauiAppBuilderExtensions
{
    public static MauiAppBuilder UseMyCustomToolkit(this MauiAppBuilder builder)
    {
        builder.UseMauiCommunityToolkit();
        builder.UseMauiCommunityToolkitCamera();

        return builder;
    }
}
```

This `MauiAppBuilderExtensions` class may (and does) live in a completely separate class library as its responsibility is to bring in the required functionality from the CommunityToolkit to other MAUI apps or class libraries that have this class library as a direct or transitive dependency. Therefore, for the purposes of this issue, you can assume that the MAUI app in question has the class library as a project reference dependency.

The bug lies in the straight fact that you cannot do this. You CAN do this with the CommunityToolkitMaui bootstrapper, but NOT the CommunityToolkitMauiCamera one. The only way I have found to initialise the camera package is by attaching the bootstrapper method directly to the MauiAppBuilder object in the MAUI project's `MauiProgram.cs` file.

For my intents and purposes, I consider this a bug.

### Expected Behavior

Keeping all of the above in mind, I should be able to register the CommunityToolkit.Maui.Camera bootstrapper method with a MauiAppBuilder that is partially constructed from an extension class in a class library outside of the current MAUI app project. I should not have to call it directly in the `MauiProgram.cs` file of the MAUI app project.

### Steps To Reproduce

Clone, open, build, and run the two sample apps provided in the linked public reproduction repository. Use the README of the repo as a guide for the behaviours being demonstrated in the samples. One sample should build and run without issues. The other will not build. Please observe the bug. The difference between the two sample apps is a matter of 3 lines. Please use the README!

### Link to public reproduction project repository

https://github.com/cam13469/CommunityCameraIssueRepro

### Environment

```markdown
- CommunityToolkit.Maui: 9.0.2
- CommunityToolkit.Maui.Camera: 1.0.3
- OS: Windows 11 Build 22631.3880 Version 23H2
- .NET MAUI: 8.0.61
```

**Comment by bijington** [comment]
I'm surprised it works for the main toolkit package and not for the camera package. 

For now as a workaround is it possible to suppress the build issue? Using this approach https://learn.microsoft.com/visualstudio/code-quality/in-source-suppression-overview?view=vs-2022&tabs=csharp

**Comment by cam13469** [comment]
Thank you for the suggestion @bijington.

## TLDR;

- Suppress analyser error `MCT001` and `MCTC001` if you want to bootstrap community toolkit packages to your MAUI app outside of the main application csproject
- If you do the above ^ give instance of your `CameraView` component in XAML an `x:Name` attribute to avoid XAML error `XFC0000` 

## Additional Info

After doing some more digging in the code responsible for the larger project I am working on, it was discovered that we are in fact supressing the build error `MCT001`, which is thrown by the community toolkit when the bootstrap method is not detected in the main MAUI application. That explains why our builds work for the MCT but not the MCT Camera package.

![Screenshot 2024-07-18 093327](https://github.com/user-attachments/assets/4a4f15cf-3fb9-4093-8af9-b283ad76427c)

After some further digging, I discovered that the analyser error being thrown by the Camera bootstrapper is given by the code `MCTC001` (note the difference!).

After suppressing this analyser error as well, I am now able to build my app.

As a side note however, I found that bringing in the CameraView as a component into my page was throwing a new XAML related error. The code that causes this can be seen below:

```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage
    x:Class="MediaSampleApp.MainPage"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:mct="http://schemas.microsoft.com/dotnet/2022/maui/toolkit">

    <ScrollView>
        <VerticalStackLayout
            Padding="30,0"
            Spacing="25"
            VerticalOptions="Center">

            <mct:CameraView />    <--! This line is causing error XFC0000 -->

        </VerticalStackLayout>
    </ScrollView>

</ContentPage>
```

I've encountered this before in other situations and I don't believe it is a fault of the MCT in any way. I found a related issue in the dotnet MAUI repo here https://github.com/dotnet/maui/issues/7503 . The simple fix for this is to give the CameraView control instance an `x:Name` attribute and then there are no issues:

```xaml
<?xml version="1.0" encoding="utf-8" ?>
<ContentPage
    x:Class="MediaSampleApp.MainPage"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:mct="http://schemas.microsoft.com/dotnet/2022/maui/toolkit">

    <ScrollView>
        <VerticalStackLayout
            Padding="30,0"
            Spacing="25"
            VerticalOptions="Center">

            <mct:CameraView x:Name="camera" />    <--! This line no longer throws error XFC0000 -->

        </VerticalStackLayout>
    </ScrollView>

</ContentPage>
```

Thanks again @bijington for your suggestion. I will close this issue after these revelations.

**Comment by bijington** [comment]
@cam13469 thabk you for the detailed update. I will try to add some of that detail into the documentation to hopefully help the next person 👍
